CalendarApp/View Controllers/ScheduleViewController.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  ScheduleViewController.swift
+//  CalendarApp
+//
+//  Created by Angelina Zhang on 7/14/22.
+//
+
+import Foundation
+import CalendarKit
+
+@objc
+public protocol ScheduleSubViewControllerDelegate {
+    func didTapEvent(_ eventID: UUID)
+    func didLongPressEvent(_ eventID: UUID)
+    func fetchEventsForDate(_ date: Date, callback: @escaping(_ events:[CalendarApp.Event]?, _ errorMessage: String?) -> Void)
+}
+
+@objcMembers
+class ScheduleSubViewController : DayViewController {
+    
+    var controllerDelegate: ScheduleSubViewControllerDelegate?
+    private var dateToCalendarKitEvents = [Date: [CalendarKit.Event]]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+    
+    func updateCalendarEvent(_ event: CalendarApp.Event, _ date: Date) {
+        guard var calendarKitEvents = dateToCalendarKitEvents[date] else {
+            return
+        }
+        let eventIndex = getEventIndex(event.objectUUID, calendarKitEvents)
+        calendarKitEvents[eventIndex] = getCalendarKitEvent(event)
+        reloadData()
+        reloadInputViews()
+    }
+    
+    func deleteCalendarEvent(_ event: CalendarApp.Event, _ date: Date) {
+        guard var calendarKitEvents = dateToCalendarKitEvents[date] else {
+            return
+        }
+        let eventIndex = getEventIndex(event.objectUUID, calendarKitEvents)
+        calendarKitEvents.remove(at: eventIndex)
+        reloadData()
+        reloadInputViews()
+    }
+    
+    func getEventIndex(_ eventID: UUID, _ calendarKitEvents: [CalendarKit.Event]) -> Int {
+        for eventIndex in 0...calendarKitEvents.count - 1 {
+            if calendarKitEvents[eventIndex].objectID == eventID {
+                return eventIndex
+            }
+        }
+        return -1
+    }
+    
+    func failedRequest(_ errorMessage: String) {
+        // TODO: Error handling
+    }
+    
+    func addCalendarKitEventFromEvents(_ events:[CalendarApp.Event], _ date: Date) {
+        for eventModel in events {
+            dateToCalendarKitEvents[date]?.append(getCalendarKitEvent(eventModel))
+        }
+    }
+    
+    func addEvent(_ eventModel: CalendarApp.Event, _ date: Date) {
+        dateToCalendarKitEvents[date]?.append(getCalendarKitEvent(eventModel))
+        reloadData()
+    }
+    
+    func getCalendarKitEvent(_ eventModel: CalendarApp.Event) -> CalendarKit.Event {
```

Comment:
Maybe `makeCalendarKitEvent` is a better name. This actually is a great time to bring up the common code smell. When you find yourself naming things with the same type name over and over, it usually means that stuff should go on its own type. 

If you have some object like:
```
struct DiningRoom {
    let size: Size
    let color: Color
    let tableType: TableType
    let tableColor: Color
}
```
At this point `DiningRoom` should just have a `Table` object with `type` and `color` fields.